processor/src/main/java/org/mapstruct/ap/internal/conversion/JavaStreamToCollectionConversion.java

```diff
@@ -41,7 +41,9 @@
 
     @Override
     protected String getToExpression(ConversionContext conversionContext) {
-        return "<SOURCE>.collect( Collectors." + collector( conversionContext ) + " )";
+        return "Optional.ofNullable( <SOURCE> )" +
```

Comment:
It doesn't help. The `TypeConversion.ftl` has `compress single_line = true` which leads to making the `\n` into spaces (At least I think it is because of that)